processing/src/main/java/io/druid/segment/data/ArrayIndexed.java

```diff
@@ -67,4 +69,9 @@ public int indexOf(T value)
   {
     return Arrays.asList(baseArray).iterator();
   }
+
+  @Override
+  public void inspectRuntimeShape(RuntimeShapeInspector inspector)
```

Comment:
It's used just for indexing, there should be no difference for Hotspot for any Object array type. So no need to distinguish between `String[]` and `Object[]`, for example.